modules/schema/src/main/java/org/apache/ignite/internal/schema/SchemaManager.java

```diff
@@ -283,9 +285,9 @@ private ColumnMapper columnMapper(
                     continue;
 
                 if (mapper == null)
-                    mapper = ColumnMapping.mapperBuilder(newDesc.length());
+                    mapper = ColumnMapping.mapperBuilder(newDesc);
 
-                mapper.add(newCol.schemaIndex(), oldCol.schemaIndex());
+                mapper.add(newCol.schemaIndex(), oldCol.schemaIndex(), null);
```

Comment:
why don't you pass `oldCol` as 3rd argument?

Reply:
Actually, the column will be ignored on merging. I wasn't able a test to get NPE here.
Maybe we should extend tests with more complex cases involving cleanup an old schema version.

I think you are right, the arg mustn't be 'null'.